src/lib.rs

```diff
@@ -581,6 +635,74 @@ impl Toast {
         ToastNotification::CreateToastNotification(&toast_xml).map_err(Into::into)
     }
 
+    /// Update progress bar title, status, progress value, progress value string
+    /// If the notification update is successful, the reason will be `NotificationUpdateResult::Succeeded`.
+    /// If the update notification fails, the reason will be `NotificationUpdateResult::Failed`.
+    /// If no notification is found, the reason will be `NotificationUpdateResult::NotificationNotFound`.
+    ///
+    /// # Example
+    /// ```rust
+    /// use std::{thread::sleep, time::Duration as StdDuration};
+    /// use tauri_winrt_notification::{Toast, Progress, NotificationUpdateResult};
+    /// 
+    /// fn main() {
+    ///     let mut progress = Progress {
+    ///         tag: "my_tag".to_string(),
+    ///         title: "video.mp4".to_string(),
+    ///         status: "Transferring files...".to_string(),
+    ///         value: 0.0,
+    ///         value_string: "0/1000 MB".to_string(),
+    ///     };
+    /// 
+    ///     Toast::new(Toast::POWERSHELL_APP_ID)
+    ///         .title("File Transfer from Phone")
+    ///         .progress(&progress)
+    ///         .show()
+    ///         .expect("Unable to send notification");
+    /// 
+    ///     for i in 1..=10 {
+    ///         sleep(StdDuration::from_secs(1));
+    ///         
+    ///         progress.value = i as f32 / 10.0;
+    ///         progress.value_string = format!("{}/1000 MB", i * 100);
+    /// 
+    ///         if i == 10 {
+    ///             progress.status = String::from("Completed");
+    ///         };
+    /// 
+    ///         if let Ok(update_result) = Toast::update_progress(Toast::POWERSHELL_APP_ID, &progress) {
+    ///             match update_result {
+    ///                 NotificationUpdateResult::Succeeded => {
+    ///                     println!("Notification updated successfully.");
+    ///                 },
+    ///                 NotificationUpdateResult::Failed => {   
+    ///                     println!("Failed to update notification")
+    ///                 },
+    ///                 NotificationUpdateResult::NotificationNotFound => {
+    ///                     println!("Notification not found. Please ensure the notification ID and Tag are correct.");
+    ///                 },
+    ///                 _ => println!("Unknown notification update result"),
+    ///             }
+    ///         };
+    ///     }
+    /// }
+    /// ```
+    pub fn update_progress(app_id: &str, progress: &Progress) -> Result<NotificationUpdateResult> {
```

Comment:
Having this method as an associate method feels weird, so instead of
```rs
Taost::new().progress(&progress).show();
Taost::update_progress(APP_ID, &progress);
```
I think we can do it like this
```rs
let toast = Toast::new().progress(&progress);
toast.show()?;
toast.set_progress(&progress)?;
```

Reply:
Thank you for the suggestion. I agree that adjusting the method design to:
```rust
let toast = Toast::new().progress(&progress);
toast.show()?;
toast.set_progress(&progress)?;
```
makes more intuitive sense. However, I believe we should still retain the update_progress method, as updating the progress bar in notifications needs to follow the approach outlined in Microsoft's documentation ([Toast Progress Bar](https://learn.microsoft.com/en-us/uwp/api/Windows.UI.Notifications.ToastNotifier.Update)). This will also keep our API consistent with Microsoft's design.
```rust
let toast = Toast::new().progress(&progress);
toast.show()?;
toast.update_progress(&progress)?;
```
or
```rust
let toast = Toast::new().set_progress(&progress);
toast.show()?;
toast.update_progress(&progress)?;
```

Reply:
This crate is not closely following the Microsoft API, it is supposed to be more familiar to Rust developers which is why this makes more sense:
```
let toast = Toast::new().progress(&progress);
toast.show()?;
toast.set_progress(&progress)?;
```

bacause `Toast::new().progress(&progress)` is a builder pattern, it doesn't need `set_` prefix which also follows the convention set by other methods like `title`, `text`, `duration`..etc

as for `set_progress` vs `update_progress` as a method, both work and both can mean the same thing but I think in general, Rust libraries tend to use `set_<thing>` instead of `update_<thing>`

Reply:
Thank you for the clarification! I understand now that the crate aims to be more familiar to Rust developers, and I see the reasoning behind using the builder pattern without the set_ prefix.

I agree that following the naming convention established by other methods makes the API more consistent and intuitive.

Thank you for your guidance on this!